# Домашняя работа №1

## Pandas

Для выполнения следующих заданий нам потребуется датасет `iris`:

In [58]:
import pandas as pd
import seaborn as sns

iris = sns.load_dataset('iris')
iris.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

### Задание 1 (0.25 балла)

Для столбца `sepal_length` посчитайте среднее значение, среднее квадратичное отклонение и медиану по каждому классу из `species`.

### Задание 2 (0.25 балла)

Посчитайте количество строк в датасете, которое удовлетворяет условию: `species == 'setosa'` и `sepal_width > 3.2`

### Задание 3 (0.5 балла)

Реализуйте функцию, которая возвращает датасет, в котором нет строк с одновременно одинаковыми значениями `sepal_width` и `species`

### Задание 4 (0.5 балла)

Создайте новую колонку `width_on_length`, в которой значение формируется согласно правилу: 

* если species - `setosa`, то значение равно `sepal_width * sepal_length`; 
* если species - `versicolor`, то значение равно `1.5 * sepal_width * sepal_length / petal_length`;
* иначе `np.nan`

### Задание 5 (0.5 балла)

Реализуйте функцию, которая заполняет пропущенные значения в колонке `width_on_length` случайными `float` значениями от 10 до 30, сортирует строки по значению `width_on_length` по убыванию и после этого перенумеровывает индекс датасета (проверьте, чтобы после этих действий не появилось новых колонок).